platforms.go

```diff
@@ -275,6 +269,11 @@ func Format(platform specs.Platform) string {
 		return "unknown"
 	}
 
+	if strings.ToLower(platform.OS) == "windows" && platform.OSVersion != "" {
```

Comment:
There was some discussion that a second `Format()` function function be needed to make sure the existing code using the old one does not start to create platforms that may not be parseable. Sorry if this was already answered, I can't locate the discussion about it.

I'm not sure about the "windows" string check here. IMHO these kind of exceptions do not belong in the format code. If the `OSVersion` is set by the user I can only assume that it should be used.

Reply:
this was needed to be added here based on the other PR feedbacks I got to combine normalizing OS and OSVersions into one single function.
cc @mikebrow

Reply:
see other comments lower down..  I do think we need one local private func that Parse() calls which passes in the combined string... and a different local private func that Format() calls that passes in the two strings that need normalization. The first normalize can call the second..

Reply:
not sure how this Format change is tied to the normalization in Parse for parts[0] which already had the os and version combined into parts[0]

Reply:
The backwards incompatible issue of adding version to the Format output is something to consider.. The user should know this is happening and opt in/out.

Reply:
![image](https://github.com/containerd/platforms/assets/99994218/242112a2-5cdc-4471-abbb-9ed50430041b)

@mikebrow @tonistiigi without the addition of the if block for windows, the format string only returned a string by joining OS, architecture and its variant. It ignored the OS version even if one was specified. So the if block was necessary to check if an OS version was specified in specs.Platform and if yes, it makes sure to add the OSVersion in braces to the OS in the first value being returned. I thought this was ok for 2.0 to change behavior? If this is not ok, I can add a FormatV2() call that user can opt in to use. Is this smth you would want me to change?

Reply:
My preference is `Format()` returning only 3 components like before and `FormatV2()` additionally returning `OSVersion` component any time it is not empty, even if `OS` is not windows.